interfaces/cython/cantera/units.pyx

```diff
@@ -1,9 +1,9 @@
 # This file is part of Cantera. See License.txt in the top-level directory or
 # at https://cantera.org/license.txt for license and copyright information.
 
-from typing import Dict
-from collections.abc import Sequence
-import numbers
+from typing import Dict as _Dict
```

Comment:
If we're supporting 3.10+, this can be changed to drop the import and use `dict` lower case.